### PR TITLE
PRMDR 402 and 403

### PR DESCRIPTION
**Pull request by RachelHowellNHS**

### CODE REVIEW

File: infrastructure/alarms.tf [hunk 1]
```diff
@@ -1,107 +1,73 @@
...
-  alarm_name        = "${terraform.workspace}_repo_5xx_alert"
-  alarm_description = "Triggers when a 5xx error is detected on the Api Gateway"
-  namespace         = "AWS/ApiGateway"
-  dimensions = {
-    ApiName = aws_api_gateway_rest_api.ndr_doc_store_api.name
-  }
-  metric_name         = "5XXError"
-  treat_missing_data  = "notBreaching"
+resource "aws_cloudwatch_metric_alarm" "api_gateway_alarm_4XX" {
+  alarm_name          = "4XX-status-${aws_api_gateway_rest_api.ndr_doc_store_api.name}"
```
Comment (NogaNHS) [inline infrastructure/alarms.tf:2]: Can you please add tags to all the new resources?

File: infrastructure/alarms.tf [hunk 2]
```diff
@@ -1,107 +1,73 @@
...
+        "Principal" : {
+          "Service" : "cloudwatch.amazonaws.com"
+        },
+        "Action" : "SNS:Publish",
+        "Condition" : {
+          "ArnLike" : {
+            "aws:SourceArn" : "arn:aws:cloudwatch:eu-west-2:${data.aws_caller_identity.current.account_id}:alarm:*"
+          }
+        }
+        "Resource" : "*"
```
Comment (NogaNHS) [inline infrastructure/alarms.tf:58]: Should this be a wildcard?

File: infrastructure/modules/kms/main.tf [hunk 3]
```diff
@@ -0,0 +1,50 @@
+resource "aws_kms_key" "encryption_key" {
+  description         = var.kms_key_description
+  policy              = data.aws_iam_policy_document.kms_key_policy_doc.json
+  enable_key_rotation = var.kms_key_rotation_enabled
+}
+
+resource "aws_kms_alias" "encryption_key_alias" {
+  name          = var.kms_key_name
+  target_key_id = aws_kms_key.encryption_key.id
+}
```
Comment (NogaNHS) [inline infrastructure/modules/kms/main.tf:10]: tags

File: infrastructure/modules/kms/main.tf [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      identifiers = ["arn:aws:iam::${var.current_account_id}:root"]
+      type        = "AWS"
+    }
+    actions   = ["kms:*"]
+    resources = ["*"]
+  }
+  statement {
+    effect = "Allow"
+    principals {
+      identifiers = ["sns.amazonaws.com"]
```
Comment (NogaNHS) [inline infrastructure/modules/kms/main.tf:29]: Looks like this is a very specific module.  Can you make this more generic?

File: infrastructure/modules/kms/main.tf (hunk 3 above)
Comment (RachelHowellNHS) [reply infrastructure/modules/kms/main.tf]: added but then removed from the alias resource as tags not allowed on aliases

**Review by NogaNHS — APPROVED** [review]

**Pull request merged**